lib/matplotlib/colors.py

```diff
@@ -742,6 +742,107 @@ def func_r(x):
 
         return LinearSegmentedColormap(name, data_r, self.N, self._gamma)
 
+    def join(self, other, name=None, frac_self=None, N=None):
```

Comment:
Maybe put a generic version of this (using `ListedColormap` in all case on `Colormap`?   Might also make sense to have a generic `join_color_map(*cmaps, fractions=None)` as a top-level function (rather than as methods)?